pkg/asset/ignition/node.go

```diff
@@ -91,3 +92,48 @@ func ConvertToAppendix(file *igntypes.File) {
 	file.Contents = igntypes.Resource{}
 	file.Overwrite = ignutil.BoolToPtr(false)
 }
+
+// AppendVarPartition appends a /var partition to the ignition configuration to avoid growfs.
+func AppendVarPartition(config *igntypes.Config) {
+	// See https://docs.fedoraproject.org/en-US/fedora-coreos/storage/#_setting_up_separate_var_mounts
+	config.Storage.Disks = append(config.Storage.Disks, igntypes.Disk{
+		Device: "/dev/disk/by-id/coreos-boot-disk",
+		Partitions: []igntypes.Partition{
+			{
+				Label:  ptr.To("root"),
+				Number: 4,
+				// Allocate at least 8GiB to the rootfs.
+				// See https://docs.fedoraproject.org/en-US/fedora-coreos/storage
+				SizeMiB: ptr.To(8192),
+				Resize:  ptr.To(true),
+			},
+			{
+				Label:   ptr.To("var"),
+				SizeMiB: ptr.To(0),
+			},
+		},
+	})
+	config.Storage.Filesystems = append(config.Storage.Filesystems, igntypes.Filesystem{
+		Path:   ptr.To("/var"),
+		Device: "/dev/disk/by-partlabel/var",
+		Format: ptr.To("xfs"),
+	})
+	// generate a mount unit so that this filesystem gets mounted in the real root.
+	config.Systemd.Units = append(config.Systemd.Units, igntypes.Unit{
+		Name:    "var.mount",
+		Enabled: ptr.To(true),
+		Contents: ptr.To(`
+[Unit]
+Requires=systemd-fsck@dev-disk-by\x2dpartlabel-var.service
+After=systemd-fsck@dev-disk-by\x2dpartlabel-var.service
+
+[Mount]
+Where=/var
+What=/dev/disk/by-partlabel/var
+Type=xfs
```

Comment:
Should add `prjquota` mount option for parity with rootfs. IIRC this is used for some of the space quota features. If you feed that Butane config from the docs into Butane, it should give you the mount unit that you can copy paste here.

Reply:
That's what I did: I ran `butane -p var.bu` where `var.bu` contents is the config in https://docs.fedoraproject.org/en-US/fedora-coreos/storage/#_setting_up_separate_var_mounts. There is no `prjquota` in the output.

Reply:
Done.